packages/wrangler/src/api/startDevWorker/utils.ts

```diff
@@ -349,23 +364,39 @@ export async function convertBindingsToCfWorkerInitBindings(
 
 	const fetchers: Record<string, ServiceFetch> = {};
 
-	for (const [name, binding] of Object.entries(inputBindings ?? {})) {
+	const iterator: [string, WorkerMetadataBinding | Binding][] = Array.isArray(
+		inputBindings
+	)
+		? inputBindings.map((b) => [b.name, b])
+		: Object.entries(inputBindings ?? {});
+
+	for (const [name, binding] of iterator) {
 		if (binding.type === "plain_text") {
 			bindings.vars ??= {};
-			bindings.vars[name] = binding.value;
+			bindings.vars[name] = "value" in binding ? binding.value : binding.text;
```

Comment:
would `binding.value ?? binding.text` work here and below?

Reply:
No, because the types don't overlap